staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation_test.go

```diff
@@ -416,6 +417,51 @@ func TestValidateCustomResource(t *testing.T) {
 				}},
 			},
 		},
+		{name: "maxProperties",
+			schema: apiextensions.JSONSchemaProps{
+				Properties: map[string]apiextensions.JSONSchemaProps{
+					"fieldX": {
+						Type:          "object",
+						MaxProperties: utilpointer.Int64(2),
+					},
+				},
+			},
+			failingObjects: []failingObject{
+				{object: map[string]interface{}{"fieldX": map[string]interface{}{"a": true, "b": true, "c": true}}, expectErrs: []string{
+					`fieldX: Too many: must have at most 2 items`,
+				}},
+			},
+		},
+		{name: "maxItems",
+			schema: apiextensions.JSONSchemaProps{
+				Properties: map[string]apiextensions.JSONSchemaProps{
+					"fieldX": {
+						Type:     "array",
+						MaxItems: utilpointer.Int64(2),
+					},
+				},
+			},
+			failingObjects: []failingObject{
+				{object: map[string]interface{}{"fieldX": []interface{}{"a", "b", "c"}}, expectErrs: []string{
+					`fieldX: Too many: must have at most 3 items`,
```

Comment:
this isn't right... kube-openapi isn't plumbing us the max here, just the actual